ttlmap.go

```diff
@@ -72,6 +72,19 @@ func New(capacity int, opts ...Option) (*TTLMap, error) {
 	return m, nil
 }
 
+// Pop removes and returns the key and the value of the oldest element
+// from the TTL Map if there are no elements in the map, returns empty
+// string and false
+func (m *TTLMap) Pop() (string, interface{}, bool) {
+	if len(m.elements) == 0 {
+		return "", nil, false
+	}
+	heapEl := m.expiryTimes.PeekEl()
+	mapEl := heapEl.Value.(*mapElement)
+	m.Remove(mapEl.key)
```

Comment:
Do you also need to remove `heapEl` from `expiryTimes`?

Reply:
m.Remove takes care of that

Reply:
Oh, right. lgtm then